app/controllers/publishers/ats_api/v1/vacancies_controller.rb

```diff
@@ -1,58 +1,79 @@
 class Publishers::AtsApi::V1::VacanciesController < Api::ApplicationController
   before_action :authenticate_client!
+  before_action :validate_payload, only: %i[create update]
+
+  rescue_from StandardError, with: :render_server_error
+  rescue_from ActiveRecord::RecordNotFound, with: :render_not_found
+  rescue_from ActionController::ParameterMissing, with: :render_bad_request
+  rescue_from Publishers::AtsApi::CreateVacancyService::InvalidOrganisationError, with: :render_bad_request
 
   def index
-    @pagy, @vacancies = pagy(vacancies, items: 100)
+    @pagy, @vacancies = pagy(vacancies.where(publisher_ats_api_client: client), items: 100)
 
     respond_to(&:json)
   end
 
   def show
-    @vacancy = vacancy
+    @vacancy = Vacancy.find(params[:id])
```

Comment:
This seems to allow any API client to fetch any vacancy in the system (theirs or not). 
Should be limited to their own vacancies:
```suggestion
    @vacancy = Vacancy.find_by!(publisher_ats_api_client: client, id: params[:id])
```